### DISCUSSION TITLE
[SOLVED] Styling a Qt application such as RSS Guard

### DISCUSSION THREAD

**Discussion by sakkamade** [martinrotter/rssguard#470] [Q&A] [answered]
So, quite an off-topic, but not too much.

I would like to have a custom stylesheet for RSS Guard (and some other Qt applications).

I found that I can use qt stylesheets like so
```bash
application -stylesheet style.qss
```

But it does not seem to work for me.
The `.qss` files _does_ work, when I use the `qt5ct` tool, but not with the command above.

Some examples of qt styling can be found [here](https://doc.qt.io/qt-5/stylesheet-reference.html).

**Comment by martinrotter** [comment, marked as answer]
Hi @sakkamade.

YES! Stylling and skining are supported by RSS Guard. Due to some advanced features used by RSS Guard I decided to encapsulate the feature as "skins" and you can use Qt stylesheets as part of the feature. In fact, as you can see in settings, there are two built in "skins".

The feature was not really documented but I know some users made some private custom skins for themselves. I created simple documentation of the feature for you.

https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md#skin

**Reply by sakkamade** [reply]
Hmm, this is not quite it. 
Well, this is also a way, huh. Thank you.

**Reply by martinrotter** [reply]
@sakkamade What do you mean?

You can just copy "vergilius" and provide your own "theme.css" file and place your "qss" code in it. It is the same thing.

**Comment by sakkamade** [comment]
I would actually like to have a single way of customising every Qt application. 
Such feature as in RSS Guard I have seen very scarcely.

I only want to understand why this doesn't work:

```
application -stylesheet style.qss
```

**Reply by martinrotter** [reply]
I believe it does not work because RSS Guard upon its start automatically loads its own default skin and thus replaces your "style.qss" passed from command line with its own "style.css".

I could probably make a fix which will NOT load RSS Guard's style once it is already loaded from command. What you think?

**Reply by sakkamade** [reply]
> I believe it does not work because RSS Guard upon its start automatically loads its own default skin and thus replaces your "style.qss"

Yes, sadly you have confirmed my fear. Thank you.

> (...) What you think?

It's fine. No need. I only wanted to understand the reason.

**Reply by sakkamade** [reply]
> It's fine. No need. I only wanted to understand the reason.

Because this is not the only Qt application which is not accepting it.

**Reply by sakkamade** [reply]
I hoped that the above command may have changed recently.

**Reply by martinrotter** [reply]
@sakkamade I made a fix for you. https://github.com/martinrotter/rssguard/commit/aa4e0d68136b8475d62db7fdb0ddd6763660c8c5

The syntax to call it for Qt apps is `application -stylesheet="path/to/stylesheet.qss"`

I tested with https://github.com/Alexhuszagh/BreezeStyleSheets

![image](https://user-images.githubusercontent.com/1255302/129847112-32be45d9-8143-447e-a199-88f4bd7c71db.png)

**Reply by sakkamade** [reply]
While the fire is burning, eh.

Nothing could stop you, huh. Oh well. Thank you!

**Comment by sakkamade** [comment]
Oh, my god! 

So, I was unable to load any stylesheets at all because of `qt5ct` tool, where I have several small custom stylesheets. 
It overrides and blocks \*any\* stylesheet I try to load application with through the terminal!

> I tested with https://github.com/Alexhuszagh/BreezeStyleSheets

And a special "Thank you" for this!
For quite a some time I was searching decent a stylesheet example.

Gosh!
